cypress/integration/networking.spec.ts

```diff
@@ -0,0 +1,65 @@
+import type { OnboardedNode } from "../plugins/index";
+
+import * as commands from "../support/commands";
+import * as path from "path";
+import { ipcStub } from "../support";
+
+context("networking", () => {
+  it("replicates a project from one node to another", () => {
+    cy.task("startNode", 17000);
+    cy.task("onboardNode", 17000);
+
+    cy.task("startNode", 18000);
+    cy.task("onboardNode", 18000);
+
+    cy.task("connectNodes", [17000, 18000]);
+
+    cy.task<OnboardedNode>("withNode", 17000).then(node => {
```

Comment:
I find that using string-based APIs on has a couple of shortcomings. The most obvious one is that we don’t get any help from TypeScript when it comes to the arguments of functions and we rely on the user to put in the return argument type. This also makes it harder to create better abstractions on top of the basic API. For example, it seems that we always run `startNode` and `onboardNode` in conjunction so maybe we want to abstract this away to make the API easier to use.

Reply:
Added cleaner separation between the plugin "backend" and "frontend". Commands are properly typed now.
https://github.com/radicle-dev/radicle-upstream/pull/1492/commits/e46870ae78be463f7244faa979625f2a267cc92b